lib/transport/tests/test_transport_factory.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2002-2018 Balabit
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "transport/transport-factory.h"
+#include "apphook.h"
+#include <criterion/criterion.h>
+
+TestSuite(transport_factory_id, .init = app_startup, .fini = app_shutdown);
+
+typedef struct _FakeTransport FakeTransport;
+typedef struct _FakeTransportFactory FakeTransportFactory;
+
+struct _FakeTransport
+{
+  LogTransport super;
+  gboolean constructed;
+};
+
+static gssize
+_fake_read(LogTransport *s, gpointer buf, gsize count, LogTransportAuxData *auc)
+{
+  return count;
+}
+
+static gssize
+_fake_write(LogTransport *s, const gpointer buf, gsize count)
+{
+  return count;
+}
+
+LogTransport *
+_fake_transport_new(void)
+{
+  FakeTransport *instance = g_new0(FakeTransport, 1);
+
+  instance->super.read = _fake_read;
+  instance->super.write = _fake_write;
+  instance->constructed = TRUE;
+
+  return &instance->super;
+}
+
+struct _FakeTransportFactory
+{
+  TransportFactory super;
+  gboolean destroyed;
+};
+
+DEFINE_TRANSPORT_FACTORY_ID_FUN(_fake_transport_factory_id);
+
+static LogTransport *
+_transport_factory_construct(const TransportFactory *s, gint fd)
+{
+  FakeTransportFactory *self = (FakeTransportFactory *)s;
```

Comment:
This cast is not needed.

Reply:
Why?
...
After a minute:
...
You are right, it is unused variable :)